File: 004Fibonacci-Swap-MinMax/Scala/fib.scala


object Fib extends App{
    
    fibo(10000)

    def fibo(last_value: Int) = {
        var num1 = 1
        var num2 = 0
        var tmp = 0

        if (num2 <= last_value){
            Console.println(num2)
            while(num1 <= last_value){
                Console.println(num1)
                tmp = num1 + num2
                num2 = num1
                num1 = tmp
            }
        }
    }

}

File: 004Fibonacci-Swap-MinMax/Java/Fib.java


class Fib{

    public static void main(String[] args){
    
        fib(10000);
    
    }

    private static void fib(int last_value){
    
        int num1 = 1, num2 = 0, tmp;
        
        if(num2 > last_value)
            return;

        System.out.println(num2);
        while(num1 < last_value){
            System.out.println(num1);
            tmp = num1 + num2;
            num2 = num1;
            num1 = tmp;
        }
    }
}



